services/telemetry/config.go

```diff
@@ -0,0 +1,64 @@
+package telemetry
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"time"
+)
+
+const (
+	distributionInfoFilePath = "/srv/pmm-distribution"
+	osInfoFilePath           = "/proc/version"
+)
+
+type Config struct {
+	Enabled      bool            `yaml:"enabled"`
+	Reporting    ReportingConfig `yaml:"reporting"`
+	Endpoints    EndpointsConfig `yaml:"endpoints"`
+	SaasHostname string          `yaml:"saas_hostname"`
+	V1URL        string          `yaml:"v1_url"`
+	V1URLEnv     string          `yaml:"v1_url_env"`
+}
+
+type EndpointsConfig struct {
+	Report string `yaml:"report"`
+}
+
+type ReportingConfig struct {
+	SkipTlsVerification bool          `yaml:"skip_tls_verification"`
+	IntervalStr         string        `yaml:"interval"`
+	IntervalEnv         string        `yaml:"interval_env"`
+	Interval            time.Duration `yaml:"-"`
+	RetryBackoffStr     string        `yaml:"retry_backoff"`
+	RetryBackoffEnv     string        `yaml:"retry_backoff_env"`
+	RetryBackoff        time.Duration `yaml:"-"`
+	SendTimeoutStr      string        `yaml:"send_timeout"`
+	SendTimeout         time.Duration `yaml:"-"`
+	RetryCount          int           `yaml:"retry_count"`
+}
+
+func (c *Config) Init() error {
+	reportingInterval, err := time.ParseDuration(c.Reporting.IntervalStr)
```

Comment:
it's weird that we parse IntervalStr here, but parse IntervalEnv in another place

Reply:
yeah, I need `time.Duration` not string. And I need ENV name, because it is in the spec...

Reply:
I mean that it's better to keep whole logic into one place

Reply:
These are independent services and configs, unifying it will make code a lot more complicated.
Implementation for V1 will be removed, so we will have no duplication.

Reply:
They overwrite the same value.
https://github.com/percona/pmm-managed/pull/1049/files#diff-7bc66d203ca22000f3fa14a2f1db4d3f013baf12a3b7930f4ad52d1f82b135bbL107-L119

I don't talk about telemetry v1 and v2, I talk about updating the same value from different places.

Reply:
If you are asking about these 2 places - it is intended. We take value from the config, which can be overridden if ENV var is set. To be consistent with requirements

Reply:
I left the comment before you moved it here. So it what I wanted 👍🏼